### PR TITLE
Move contribution guideline to contributing.md and reference from readme

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,19 @@ contributors.
...
+We recommend that any PR (unless it is trivial) to be smaller than
+500 lines (excluding go mod/sum changes).
+When adding a new component consider submitting different PRs for:
+
+* First PR should include the overall structure of the new component:
+  * Readme, configuration, and factory implementation usually using the helper factory structs.
+  * This PR is usually trivial to review, so the size limit does not apply to it.
+* Second PR should include the concrete implementation of the component.
+If the size of this PR is larger than the recommended size consider splitting it in multiple PRs.
+* Last PR should be to enable the new component and add it to the `otelcontribcol` binary.
```
Comment (tigrannajaryan) [inline README.md:110]: I suggest also adding: "The component should be enabled only after sufficient tests and code coverage are attained and there is enough confidence in the stability and quality of the component."

File: README.md [hunk 2]
```diff
@@ -95,6 +95,19 @@ contributors.
   needed.
 - Add a `replace` directive at the root `go.mod` file so your component is
   included in the build of the contrib executable.
+  
+### How to get proper reviews?
+
+We recommend that any PR (unless it is trivial) to be smaller than
+500 lines (excluding go mod/sum changes).
```
Comment (tigrannajaryan) [inline README.md:102]:
```suggestion
500 lines (excluding go mod/sum changes) in order to help reviewers to do a thorough and reasonably fast reviews.
```

**Review by tigrannajaryan — COMMENTED** [review]
Good initiative. Added a couple additional suggestions.

File: README.md (hunk 1 above)
Comment (james-bebbington) [reply README.md:110]: Should also specify what "enabling" the component means. Do you mean adding it to `components.go`?

File: README.md [hunk 3]
```diff
@@ -95,6 +95,19 @@ contributors.
   needed.
 - Add a `replace` directive at the root `go.mod` file so your component is
   included in the build of the contrib executable.
+  
+### How to get proper reviews?
```
Comment (james-bebbington) [inline README.md:99]:
```suggestion
### How to structure PRs to get expedient reviews?
```

File: README.md (hunk 1 above)
Comment (pmcollins) [reply README.md:110]: Maybe a note about refactoring? I'm assuming refactoring should generally go into a separate PR?

**Comment by bogdandrutu** [comment]
Per offline discussion will move this to core and link from contrib

**Comment by bogdandrutu** [comment]
New core PR https://github.com/open-telemetry/opentelemetry-collector/pull/1755

**Comment by tigrannajaryan** [comment]
Core PR merged. @bogdandrutu do you want to close this PR or change it to refer to core instructions?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,5 +1,63 @@
-## Contributing
+# Contributing
 
 If you would like to contribute please read OpenTelemetry Collector
 [contributing guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md)
 before you begin your work.
+
+## Adding New Components
+Before you start please read the [contributing
+guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md).
```
Comment (tigrannajaryan) [inline CONTRIBUTING.md:9]: Duplicate information. We already have this a few lines above. Eliminate one of the paragraphs.

Comment (bogdandrutu) [reply CONTRIBUTING.md]: done.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**